### ISSUE TITLE
Add HealthCheck "name" in CWMMH0052W error message to help investigation

### ISSUE THREAD

**Issue by Kiiv** [OpenLiberty/open-liberty#28048]
**Describe the use case that you want to enable:**
When a healthcheck report a down status, a WARN log is generated to give information about what is going on. The message is :
```
healthcheck.application.down.CWMMH0052W=CWMMH0052W: The {0} implementing HealthCheckResponse in the {1} application in module {2}, reported a {3} status with data {4}.
```
(from [this file](https://github.com/OpenLiberty/open-liberty/blob/3ff6f11f3a68872e0af176fde71ec34d42fd706c/dev/io.openliberty.microprofile.health.internal.common/resources/io/openliberty/microprofile/health/resources/Health.nlsprops)

In real condition this give something like :
```
[3/29/24, 11:37:42:554 CET] 00000036 icroprofile.health31.services.impl.HealthCheck31ExecutorImpl W CWMMH0052W: The class io.openliberty.microprofile.health30.impl.HealthCheck30ResponseImpl implementing HealthCheckResponse in the XXX-app application in module XXX-app-1.0.0.war, reported a DOWN status with data Optional[{ERROR=Liveness checks not found}].
```

The message contains the custom data that has been added in the healthcheck implementation in case of error but not the "name" attribute of the HealthCheckResponse.
So there is no information about which check is in error.

**Describe why this is important to you:**
It should be great to have more context in the log message when a healthcheck return DOWN to be able to investigate. Sometime we need to find the cause when the problem is not there anymore. In that case it's not possible to call the healthcheck url to get the full context.

**Additional context**
It's possible to add that information manually in the data but it's more work for something that could already be traced easily (I hope :)).

**Comment by pgunapal** [comment]
Hi @Kiiv Yes, this is a good suggestion. We will work to get the message updated to include the HealthCheck "name" attribute. Will update the issue, once a fix is merged. Thank you!

**Comment by pgunapal** [comment]
The PR with the fix for this issue was just merged, it should be targeted to be included in fixpack 24.0.0.6
